pkg/interactor/list_data.go

```diff
@@ -10,94 +10,55 @@ import (
 
 // List provides all the thing's data owned by an user
 func (d *DataInteractor) List(token string, query *entities.Query) ([]entities.Data, error) {
-	data, err := d.getDevicesData(token, query)
+	things, err := d.things.List(token)
 	if err != nil {
-		return []entities.Data{}, err
+		return nil, fmt.Errorf("error getting list of things: %w", err)
 	}
 
-	data, err = d.filterDataBySensorID(data, query.ThingID, query.SensorID)
-	if err != nil {
-		return []entities.Data{}, err
+	if query.ThingID != "" {
+		things = []*btEntities.Thing{{ID: query.ThingID}}
 	}
 
-	return data, err
-}
-
-func (d *DataInteractor) getDevicesData(token string, query *entities.Query) ([]entities.Data, error) {
-	data := []entities.Data{}
-	things, err := d.things.List(token)
+	data, err := d.getAllData(things, query)
 	if err != nil {
-		return data, fmt.Errorf("%v: %w", ErrValidToken, err)
+		return nil, fmt.Errorf("error getting the data: %w", err)
 	}
 
-	if query.ThingID != "" {
-		err := d.verifyAuthorization(token, query.ThingID)
+	if query.SensorID != "" {
```

Comment:
I would prefer not to come back to a less modular code in comparison with the current code. So, if it's necessary to refactor this interactor let's do it in another commit apart from the specific bug fix, and let's keep it as clean as possible.

Reply:
Let me know if you have any questions.

Reply:
Part of this refactor was necessary exactly for fix the listing data bug. I remove some functions since it appears to be simple to use inside the `List` main function, but I could add them back

Reply:
No problem, I will wait for you with respect to the outcome of the tests so that we can decide what to do.